File: app/src/main/java/net/twisterrob/colorfilters/android/PreferencesActivity.java
package net.twisterrob.colorfilters.android;

import android.annotation.TargetApi;
import android.os.*;
import android.preference.PreferenceActivity;

public class PreferencesActivity extends PreferenceActivity {
	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		addPreferencesFromResource(R.xml.preferences);
	}

	@TargetApi(Build.VERSION_CODES.KITKAT)
	@Override
	protected boolean isValidFragment(String fragmentName) {
		return false;
	}
}

File: app/src/main/kotlin/net/twisterrob/colorfilters/android/PreferencesActivity.kt
package net.twisterrob.colorfilters.android

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceFragmentCompat

class PreferencesActivity : AppCompatActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_settings)
		if (savedInstanceState == null) {
			supportFragmentManager
				.beginTransaction()
				.add(R.id.prefs_container, PreferencesFragment())
				.commit()
		}
	}

	class PreferencesFragment : PreferenceFragmentCompat() {
		override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
			setPreferencesFromResource(R.xml.preferences, rootKey)
		}
	}
}
